### PR TITLE
chore: Improve test utils

### PR DESCRIPTION
**Pull request by marikaner**
Improve test-utils to avoid confusion between v2/v4 entities in tests. This also fixes some incorrect usage of v2/v4 entites.

### CODE REVIEW

File: packages/odata-common/src/request-builder/batch/batch-request-builder.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ import { serializeBatchRequest } from './batch-request-serializer';
  * @internal
  */
 export class BatchRequestBuilder<
-  DeSerializersT extends DeSerializers
```
Comment (FrankEssenberger) [inline packages/odata-common/src/request-builder/batch/batch-request-builder.ts:22]: scope creep :-)

**Review by FrankEssenberger — APPROVED** [review]
LGTM

File: packages/odata-common/src/request-builder/batch/batch-request-builder.ts (hunk 1 above)
Comment (marikaner) [reply packages/odata-common/src/request-builder/batch/batch-request-builder.ts:22]: whoopsie daisy, this was not on purpose

File: packages/odata-common/src/request-builder/batch/batch-request-builder.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ import { serializeBatchRequest } from './batch-request-serializer';
...
+  DeSerializersT extends DeSerializers = DefaultDeSerializers
```
Comment (marikaner) [inline packages/odata-common/src/request-builder/batch/batch-request-builder.ts:22]:
```suggestion
  DeSerializersT extends DeSerializers
```

File: packages/odata-common/src/request-builder/batch/batch-request-builder.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ import {
 import { HttpResponse } from '@sap-cloud-sdk/http-client';
 import { MethodRequestBuilder } from '../request-builder-base';
 import { ODataBatchRequestConfig, ODataRequest } from '../../request';
-import { DeSerializers } from '../../de-serializers';
+import { DefaultDeSerializers, DeSerializers } from '../../de-serializers';
```
Comment (marikaner) [inline packages/odata-common/src/request-builder/batch/batch-request-builder.ts:8]:
```suggestion
import { DeSerializers } from '../../de-serializers';
```
